src/ECharts/Formatter.purs

```diff
@@ -15,10 +16,14 @@ import Control.Monad.Eff
 
 type FormatParams = Json
 
+foreign import data GenericFormatter :: *
+
 data Formatter =
   Template String
   | FormatFunc (forall eff. Array FormatParams -> Eff eff String)
-
+  | StringFormatFunc (String -> String)
+  | NumberFormatFunc (Number -> String)
+  | GenericFormatFunc GenericFormatter
```

Comment:
Changed  the foreign javascript function `toolTipFomatter` to purescript function